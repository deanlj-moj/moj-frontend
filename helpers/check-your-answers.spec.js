const checkYourAnswers = require('./check-your-answers')

// Sample session data where all values are objects
const session = {
  '/accessibility-findings': { field1: 'Field Value', field2: 'Another Value' },
  '/accessibility-findings-more': { detail: 'Some additional information' },
  '/additional-information': { info: 'Some additional info' },
  '/component-code': { code: '1234', language: 'JavaScript' },
  '/component-code-details': { detail: 'Extra details' },
  '/component-image': { image: 'image.png' },
  '/prototype': { data: 'Prototype data' },
  '/prototype-url': { url: 'http://example.com' },
  '/your-details': { name: 'Sarah Philips', email: 'sarah@example.com' }
}

describe('checkYourAnswers function', () => {
  it('should transform session data into the expected structure', () => {
    const result = checkYourAnswers(session)

    // Snapshot of the result
    expect(result).toMatchSnapshot()
  })

  it('should handle missing session fields gracefully', () => {
    const incompleteSession = {
      '/accessibility-findings': { field1: 'Field Value' }, // field2 missing
      '/component-code': { code: '1234' },
      '/prototype': { data: 'Prototype data' }
      // Missing some fields like 'component-image', 'your-details'
    }

    const result = checkYourAnswers(incompleteSession)

    // Snapshot of the result with missing fields
    expect(result).toMatchSnapshot()
  })

  it('should handle empty session gracefully', () => {
    const emptySession = {}
    const result = checkYourAnswers(emptySession)

    // Snapshot of the result with an empty session
    expect(result).toMatchSnapshot()
  })
})
