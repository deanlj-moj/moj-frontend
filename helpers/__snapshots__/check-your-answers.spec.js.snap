// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkYourAnswers function should handle empty session gracefully 1`] = `
{
  "accessibilityFindingsRows": [],
  "componentCodeRows": [],
  "componentDetailsRows": [],
  "componentImageRows": [],
  "prototypeRows": [],
  "yourDetailsRows": [],
}
`;

exports[`checkYourAnswers function should handle missing session fields gracefully 1`] = `
{
  "accessibilityFindingsRows": [
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/accessibility-findings",
            "text": "Change",
            "visuallyHiddenText": "Accessibility Findings - Field1",
          },
        ],
      },
      "key": {
        "text": "Field1",
      },
      "value": {
        "text": "Field Value",
      },
    },
  ],
  "componentCodeRows": [
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/component-code",
            "text": "Change",
            "visuallyHiddenText": "Component Code - Code",
          },
        ],
      },
      "key": {
        "text": "Code",
      },
      "value": {
        "text": "1234",
      },
    },
  ],
  "componentDetailsRows": [],
  "componentImageRows": [],
  "prototypeRows": [
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/prototype",
            "text": "Change",
            "visuallyHiddenText": "Prototype - Data",
          },
        ],
      },
      "key": {
        "text": "Data",
      },
      "value": {
        "text": "Prototype data",
      },
    },
  ],
  "yourDetailsRows": [],
}
`;

exports[`checkYourAnswers function should handle multiple entries for a field 1`] = `
{
  "accessibilityFindingsRows": [],
  "componentCodeRows": [],
  "componentDetailsRows": [],
  "componentImageRows": [],
  "prototypeRows": [],
  "yourDetailsRows": [],
}
`;

exports[`checkYourAnswers function should transform session data into the expected structure 1`] = `
{
  "accessibilityFindingsRows": [
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/accessibility-findings",
            "text": "Change",
            "visuallyHiddenText": "Accessibility Findings - Field1",
          },
        ],
      },
      "key": {
        "text": "Field1",
      },
      "value": {
        "text": "Field Value",
      },
    },
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/accessibility-findings",
            "text": "Change",
            "visuallyHiddenText": "Accessibility Findings - Field2",
          },
        ],
      },
      "key": {
        "text": "Field2",
      },
      "value": {
        "text": "Another Value",
      },
    },
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/accessibility-findings-more",
            "text": "Change",
            "visuallyHiddenText": "Accessibility Findings More - Detail",
          },
        ],
      },
      "key": {
        "text": "Detail",
      },
      "value": {
        "text": "Some additional information",
      },
    },
  ],
  "componentCodeRows": [
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/component-code",
            "text": "Change",
            "visuallyHiddenText": "Component Code - Code",
          },
        ],
      },
      "key": {
        "text": "Code",
      },
      "value": {
        "text": "1234",
      },
    },
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/component-code",
            "text": "Change",
            "visuallyHiddenText": "Component Code - Language",
          },
        ],
      },
      "key": {
        "text": "Language",
      },
      "value": {
        "text": "JavaScript",
      },
    },
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/component-code-details",
            "text": "Change",
            "visuallyHiddenText": "Component Code Details - Detail",
          },
        ],
      },
      "key": {
        "text": "Detail",
      },
      "value": {
        "text": "Extra details",
      },
    },
  ],
  "componentDetailsRows": [],
  "componentImageRows": [
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/component-image",
            "text": "Change",
            "visuallyHiddenText": "Component Image - Image",
          },
        ],
      },
      "key": {
        "text": "Image",
      },
      "value": {
        "text": "image.png",
      },
    },
  ],
  "prototypeRows": [
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/prototype",
            "text": "Change",
            "visuallyHiddenText": "Prototype - Data",
          },
        ],
      },
      "key": {
        "text": "Data",
      },
      "value": {
        "text": "Prototype data",
      },
    },
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/prototype-url",
            "text": "Change",
            "visuallyHiddenText": "Prototype Url - Url",
          },
        ],
      },
      "key": {
        "text": "Url",
      },
      "value": {
        "text": "http://example.com",
      },
    },
  ],
  "yourDetailsRows": [
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/your-details",
            "text": "Change",
            "visuallyHiddenText": "Your Details - Name",
          },
        ],
      },
      "key": {
        "text": "Name",
      },
      "value": {
        "text": "Sarah Philips",
      },
    },
    {
      "actions": {
        "items": [
          {
            "href": "/get-involved/add-new-component/your-details",
            "text": "Change",
            "visuallyHiddenText": "Your Details - Email",
          },
        ],
      },
      "key": {
        "text": "Email",
      },
      "value": {
        "text": "sarah@example.com",
      },
    },
  ],
}
`;
